tests/unit/test_check_metrics.py

```
@@ -0,0 +1,407 @@
+        """Test with two values."""
+        data = {"1": 1.0, "2": 10.0}
+        result = mean(data, ignore_top_p=0.5)
+        # Should remove top 50% (1 value), leaving just 1.0
+        assert result == 1.0
+
+    def test_mean_ignore_top_p_invalid_range(self):
+        """Test that invalid ignore_top_p values raise an error."""
+        data = {"1": 1.0, "2": 2.0, "3": 3.0}
+
+        with pytest.raises(
+            ValueError, match="ignore_top_p must be between 0.0 and 1.0"
```

Ruff (0.14.0)

Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)